### PR TITLE
OverlayNG: Interpolate M values

### PR DESCRIPTION
**Pull request by dbaston**
This PR adds support for interpolation of M values in OverlayNG. M values from the input are preserved and M values at constructed points are populated via linear interpolation if possible. Unlike the Z dimension, the `ElevationModel` is not used to populate the M value of output coordinates that do not intersect segments with M values.

This does not cover all cases - for example, if snap-rounding is required, I expect that M values will be lost. I will work on adding test cases so that these usages can be covered.

By not assuming the use of the `Coordinate` type, these changes also move further in the direction of using XY inputs without carrying around a Z value.

### CODE REVIEW

**Comment by mwtoews** [comment]
Also [`web/content/usage/download.md`](https://github.com/libgeos/geos/blob/main/web/content/usage/download.md) will need to be updated to mention "C++14 compiler".

File: CMakeLists.txt [hunk 1]
```diff
@@ -128,7 +128,7 @@ include(CMakeDependentOption)
 
 ## CMake global variables
 option(BUILD_SHARED_LIBS "Build GEOS with shared libraries" ON)
-set(CMAKE_CXX_STANDARD 11 CACHE STRING "C++ standard version to use (default is 11)")
+set(CMAKE_CXX_STANDARD 14 CACHE STRING "C++ standard version to use (default is 11)")
```
[the remark this answers is not in the document]
Comment (dbaston) [reply CMakeLists.txt:131]: Addressed in edf5503b52c970f14f3e46f80b0d9a1c24123962, thanks

**Comment by dbaston** [comment]
@pramsey @dr-jts Are you comfortable with the semantics used for M values here? ("Interpolate like Z but don't use the ElevationModel")

**Comment by dr-jts** [comment]
> Are you comfortable with the semantics used for M values here? ("Interpolate like Z but don't use the ElevationModel")

Yes, seems reasonable.

**Comment by dbaston** [comment]
Performance is looking OK for the XY(Z) cases. Maybe a small regression in intersection of inputs that are already fully noded (triangulation results, world.wkt).

image

**Pull request merged**